homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,223 @@
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        self._errors = {}
+
+        if user_input is not None:
+            self._data[CONF_HOST] = user_input[CONF_HOST]
+            if self._host_in_configuration_exists(self._data[CONF_HOST]):
+                self._errors[CONF_HOST] = "already_configured"
+            self._data[CONF_NAME] = user_input[CONF_NAME]
+            if await self._try_connect(self._data[CONF_HOST]):
+                await self.async_set_unique_id(self._get_serial_no())
+                return await self.async_step_inverter()
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_HOST] = DEFAULT_HOST
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input[CONF_HOST]): cv.string,
+                    vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): cv.string,
+                }
+            ),
+            errors=self._errors,
+            last_step=False,
+        )
+
+    async def async_step_inverter(self, user_input=None):
+        """Give the user the opportunities to override inverter config."""
+
+        self._errors = {}
+        if user_input is not None:
+            if not (
+                (0 <= user_input[CONF_CAPABILITY] <= 7)
+                or (256 <= user_input[CONF_CAPABILITY] <= 263)
+            ):
+                self._errors[CONF_CAPABILITY] = "capabilty_invalid"
+            else:
+                self._data[CONF_CAPABILITY] = user_input[CONF_CAPABILITY]
+                self._data[CONF_MAX_OUTPUT] = user_input[CONF_MAX_OUTPUT]
+                self._data[CONF_DC_STRINGS] = user_input[CONF_DC_STRINGS]
+                title = self._get_name()
+                return self.async_create_entry(title=title, data=self._data)
+        else:
+            user_input = {}
+            user_input[CONF_MAX_OUTPUT] = self._get_max_output()
+            user_input[CONF_DC_STRINGS] = self._get_dc_strings()
+            user_input[CONF_CAPABILITY] = self._get_capability()
+
+        return self.async_show_form(
+            step_id="inverter",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_MAX_OUTPUT, default=user_input[CONF_MAX_OUTPUT]
+                    ): vol.In([1.5, 2.0, 3.0, 3.8, 4.0, 5.0]),
+                    vol.Required(
+                        CONF_DC_STRINGS, default=user_input[CONF_DC_STRINGS]
+                    ): vol.In([1, 2]),
+                    vol.Required(
+                        CONF_CAPABILITY, default=user_input[CONF_CAPABILITY]
+                    ): int,
+                }
+            ),
+            errors=self._errors,
+            last_step=True,
+        )
+
+    async def async_step_import(self, user_input=None):
```

Comment:
You don't need this. Imports are to handle `configuration.yaml` entries so that wouldn't apply here